pkg/ceph/osd/health_test.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2017 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package osd
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/rook/rook/pkg/ceph/mon"
+	"github.com/rook/rook/pkg/clusterd"
+	exectest "github.com/rook/rook/pkg/util/exec/test"
+	"github.com/rook/rook/pkg/util/kvstore"
+	"github.com/rook/rook/pkg/util/proc"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTAgent(t *testing.T, devices, configDir string, storeConfig *StoreConfig) (*OsdAgent, *exectest.MockExecutor) {
+	location := "root=here"
+	forceFormat := false
+	if storeConfig == nil {
+		storeConfig = &StoreConfig{StoreType: "bluestore"}
+	}
+	agent := NewAgent(devices, false, "", "", forceFormat, location, *storeConfig, nil, "myhost", kvstore.NewMockKeyValueStore())
+	agent.cluster = &mon.ClusterInfo{Name: "myclust"}
+
+	executor := &exectest.MockExecutor{
+		MockExecuteCommandWithOutputFile: func(debug bool, actionName string, command string, outFileArg string, args ...string) (string, error) {
+			return "{\"key\":\"mysecurekey\", \"osdid\":3.0}", nil
+		},
+	}
+
+	return agent, executor
+}
+
+func TestOSStatusRuns(t *testing.T) {
+	configDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("failed to create temp config dir: %+v", err)
+	}
+	defer os.RemoveAll(configDir)
+
+	storeConfig := &StoreConfig{StoreType: "bluestore"}
+
+	agent, executor := createTAgent(t, "sdx,sdy", configDir, storeConfig)
+	executor.MockExecuteCommandWithOutputFile = func(debug bool, actionName string,
+		command string, outFileArg string, args ...string) (string, error) {
+		logger.Infof("ExecuteCommandWithOutput: %s %v", command, args)
+		if args[0] == "osd" && args[1] == "dump" {
+			return `{"OSDs": [{"OSD": 0, "Up": 0, "In": 1}]}`, nil
+		}
+		return "", nil
+	}
+	context := &clusterd.Context{Executor: executor, ProcMan: proc.New(executor)}
+	mon := NewMonitor(context, agent)
+
+	err = mon.osdStatus()
```

Comment:
it would be also great to add some sort of check that verifies Stop() was actually called on the down OSD.  that would be really useful to tell if that logic is actually working.  Note that you may need to make the procMan a mockable interface and definitely inject the `osdGracePeriod` and `healthCheckInterval` to much smaller values so the test doesn't take 600 seconds to run :)